EIPS/eip-7160.md

```diff
@@ -0,0 +1,163 @@
+---
+eip: 7160
+title: ERC-721 Multi-Metadata Extension
+description: "This EIP proposes an extension to the ERC-721 to support multiple metadata URIs per token."
+author: 0xG (@0xGh), Marco Peyfuss (@mpeyfuss)
+discussions-to: https://ethereum-magicians.org/t/erc721-multi-metadata-extension/14629
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-06-09
+requires: 165, 721
+---
+
+## Abstract
+
+This EIP proposes an extension to the [ERC-721](./eip-721.md) standard to support multiple metadata URIs per token. It introduces a new interface, `IERC721MultiMetadata`, which provides methods for accessing the metadata URIs associated with a token, including a pinned URI index and a list of all metadata URIs. The extension is designed to be backward compatible with existing `ERC721Metadata` implementations.
+
+## Motivation
+
+The current [ERC-721](./eip-721.md) standard allows for a single metadata URI per token. However, there are use cases where multiple metadata URIs are desirable, such as when a token represents a collection of (cycling) assets with individual metadata, historic token metadata, collaborative and multi-artist tokens, evolving tokens. This extension enables such use cases by introducing the concept of multi-metadata support.
+
+## Specification
+
+The `IERC721MultiMetadata` interface extends the existing `IERC721` interface and introduces additional methods and events:
+
+```solidity
+interface IERC721MultiMetadata is IERC721 {
```

Comment:
I'm no Solidity expert, but I've been told to avoid making interfaces depend on one another:

```suggestion
interface IERC721MultiMetadata /* is IERC721 */ {
```

Reply:
Commented out as suggested. By the way should the interface name reflect the EIP name instead of being called `IERC721MultiMetadata` or is it fine to give it a name?

Reply:
You can name it however you like!